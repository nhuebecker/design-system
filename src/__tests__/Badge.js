import React from 'react'
import renderer from 'react-test-renderer'
import { Badge, theme } from '..'

describe('Badge', () => {
  test('renders', () => {
    const json = renderer.create(<Badge />).toJSON()
    expect(json).toMatchSnapshot()
  })

  test('accepts custom bg and color', () => {
    const json = renderer.create(<Badge bg="white" color="accent" />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.colors.white)
    expect(json).toHaveStyleRule('color', theme.colors.accent)
  })
})
