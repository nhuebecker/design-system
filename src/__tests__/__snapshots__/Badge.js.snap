// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge accepts custom bg and color 1`] = `
.c1 {
  font-size: 16px;
  margin: 0px;
  padding-left: 12px;
  padding-right: 12px;
  padding-top: 6px;
  padding-bottom: 6px;
  background-color: #ffffff;
  color: #7280ed;
}

.c0 {
  border-radius: 4px;
  display: inline-block;
  font-weight: normal;
  -webkit-letter-spacing: 0.0375em;
  -moz-letter-spacing: 0.0375em;
  -ms-letter-spacing: 0.0375em;
  letter-spacing: 0.0375em;
  line-height: 1.25;
  text-transform: uppercase;
}

<span
  className="c0 c1"
  color="accent"
  fontSize={1}
/>
`;

exports[`Badge renders 1`] = `
.c1 {
  font-size: 16px;
  margin: 0px;
  padding-left: 12px;
  padding-right: 12px;
  padding-top: 6px;
  padding-bottom: 6px;
  background-color: #e42d42;
  color: #ffffff;
}

.c0 {
  border-radius: 4px;
  display: inline-block;
  font-weight: normal;
  -webkit-letter-spacing: 0.0375em;
  -moz-letter-spacing: 0.0375em;
  -ms-letter-spacing: 0.0375em;
  letter-spacing: 0.0375em;
  line-height: 1.25;
  text-transform: uppercase;
}

<span
  className="c0 c1"
  color="white"
  fontSize={1}
/>
`;
